cdap-proto/src/main/java/io/cdap/cdap/proto/security/Principal.java

```diff
@@ -39,7 +39,7 @@
   private final String name;
   private final PrincipalType type;
   private final String kerberosPrincipal;
-  private final String credential;
+  private final Credential credential;
```

Comment:
That's a proto change. Are there any compatibility considerations here? @albertshau

Reply:
I think it should be fine since we control all services which use Principal in the codebase, though I'll keep the thread open so @albertshau can chime in.

Reply:
@tivv and I discussed offline, decided we will add a new method to maintain backwards compatibility for the Principal proto

Reply:
I'm not familiar with which APIs this gets used with, but we need to maintain compatibility for any user facing API (those exposed by the router). If this is only used in an internal API, we can technically change it, but it's better practice to keep it compatible as well.

Reply:
The Principal is part of the security-spi I believe and is used by authorization extensions. We figure it's appropriate in this case to maintain backwards compatibility.